keychain_mac.cpp

```diff
@@ -221,7 +221,7 @@ void deletePassword(const std::string &package, const std::string &service,
         }
     }
 
-    if (item) {
+    if (!err && item) {
```

Comment:
This fixes the error observed [during this CI run](https://github.com/hrantzsch/keychain/pull/19/checks?check_run_id=370613055). It is not clear to me why the segfault occurs here but not on master. It occurs in both Debug and Release configuration. Maybe it's the `-O0` setting or some other coverage-related compiler flag.

Reply:
Can you push this out as a patch release? Potentially independent from this PR?
We're planning to integrate the library in our next release and would be glad to have this fixed beforehand.

Reply:
Definitely. We could also not wait for the Windows coverage and merge it once we're happy with the two working platforms.